lib/ansible/modules/cloud/azure/azure_rm_image.py

```diff
@@ -0,0 +1,365 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2017 Yuwei Zhou, <yuwzho@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_image
+version_added: "2.5"
+short_description: Manage Azure image.
+description:
+    - Create, delete an image from virtual machine, blob uri, managed disk or snapshot.
+options:
+    resource_group:
+        description:
+            - Name of resource group.
+        required: true
+    name:
+        description:
+            - Name of the image.
+        required: true
+    source:
+        description:
+            - OS disk source from the same region, including a virtual machine id or name,
+              OS disk blob uri, managed OS disk id or name, or OS snapshot id or name.
+        required: true
+    data_disk_sources:
+        description:
+            - List of data disk sources, including unmanaged blob uri, managed disk id or name, or snapshot id or name.
+    location:
+        description:
+            - Location of the image.
```

Comment:
Should probably state that if not specified, the location is derived from the resource group location